### ISSUE TITLE
BarcodeScanner Orientation off on iOS

### ISSUE THREAD

**Issue by mslavik2** [expo/expo#5332]
### Environment

Snack using sdk 34.0.0 and latest iOS client (2.12.3.106566)

### Steps to Reproduce

Use the BarcodeScanner on iOS

### Expected Behavior

The BarcodeScanner video to rotate properly on iOS devices.

### Actual Behavior

Camera video in the BarcodeScanner rotates incorrectly. When rotating in one direction it rotates the opposite direction.

### Reproducible Demo

https://snack.expo.io/@documentation/barcodescanner-example

**Comment by cruzach** [comment]
Hm..I tried this on the same client, both SDK 33 and SDK 34 and saw the correct rotation. 

Is this happening on a specific device? Maybe iPad, I'm guessing?

**Comment by mslavik2** [comment]
Re-tested it on iPad (iOS 12.4) and iPad mini (iOS 12.3.1) both saw the issue. Tested on an iPhone (iOS 12.2 & iOS 12.4) and it rotated correctly. Looks to be an iPad issue.

**Comment by cruzach** [comment]
Yeah, there seem to be issues with screen orientation on iPad- see #5339

**Comment by cruzach** [comment]
@mslavik2, looks like the Orientation issue on iPad's only happens in the Expo Client app. Could you test this in a standalone IPA and let me know if it's the same behavior? 

To run it on your iOS Simulator, first build your expo project with the simulator flag by running `expo build:ios -t simulator`, then download the tarball with the link given upon completion. Unpack the tar.gz by running `tar -xvzf your-app.tar.gz`. Then you can run it by starting an iOS Simulator instance, then running `xcrun simctl install booted <app path>` and then opening the installed app

**Comment by mslavik2** [comment]
@cruzach I cant seem to get the camera to show in the simulator so I am going to have to make a testflight build which will take a bit to get done. I'll update here with my findings.

**Comment by mslavik2** [comment]
@cruzach After getting a build on testflight, the orientation of the barcode scanner is correct in a standalone build.
